Up H4s to H3 in FAQ (#394)
returntocorp/semgrep-docs: docs/faq.md

import TOCInline from "@theme/TOCInline"

# Frequently Asked Questions

<TOCInline toc={toc} />

## General

### How are Semgrep and its rules licensed?

The [Semgrep command line tool](https://github.com/returntocorp/semgrep) is open-source, licensed under [GPL v3](<https://tldrlegal.com/license/gnu-general-public-license-v3-(gpl-3)>) — you can use it at work, on private and proprietary code, no problem!

The [Semgrep Registry](https://semgrep.dev/explore) contains rules from different contributors. Most rules, including all community-contributed rules, are under [MIT license](https://tldrlegal.com/license/mit-license). Some of the Registry rules written by r2c are licensed under the [Semgrep Registry license](../licensing/#semgrep-registry-license).

The source for many r2c-written Registry rules is available at [returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/). Those licensed under the Semgrep Registry license can’t be resold without r2c’s permission. Since r2c offers a paid, hosted application (Semgrep App), it’s important to have this restriction so other companies, like major cloud providers, can’t just resell our rules as a competing service.

> ### Is it ok to run Semgrep or the r2c rules on my work projects?

Reviewer: ```suggestion
### It's ok and safe to run Semgrep or the r2c rules on my work projects
```

Yes! Semgrep is safe to run on your private code. The [Semgrep Registry license’s](../licensing/#semgrep-registry-license) commercial restrictions only come into effect if you are _selling_ the rules provided in the semgrep-rules repository. If that’s the case, you’ll need to talk with r2c first to get permission.

### I’m a security professional and want to use the semgrep-rules project with my clients as part of my paid services. Is that ok?

If you have a typical consulting service and running `semgrep` or `semgrep-rules` is part of your assessments, that’s great and of course feel free to refer your clients to the [Semgrep App](https://semgrep.dev/). But if your entire service is about scanning code and you want to charge for running the `semgrep-rules` repository that r2c and the community created and maintains, you need to discuss this with r2c.

> ### What is your support policy?

Reviewer: ```suggestion
### Understanding r2c support policy
```

Help is available for all users, free or otherwise, through the [r2c Community Slack](https://r2c.dev/slack). Semgrep Team tier customers receive 8\*5 email/phone/Slack support with committed SLAs. See [Support](../support/) for more details.

> ### Can I embed the Playground in my website or blog post?

Reviewer: ```suggestion
### Embedding the Playground in website or blog post
```

> Yes! You can embed a special version of Semgrep Playground with an `iframe`. The source is `https://semgrep.dev/embed/editor?snippet=<snippet-id>` where the `snippet-id` is either the short identifier generated when you share a Playground link (this usually looks like `DzKv`) or the named identifier from a saved rule (this usually looks like `username:rule-name`).

Reviewer: ```suggestion
To embed the Playground in a website or a blog post using a special version of Semgrep Playground with an iframe. The source is `https://semgrep.dev/embed/editor?snippet=<snippet-id>` where the `snippet-id` is either the short identifier generated when you share a Playground link (this usually looks like `DzKv`) or the named identifier from a saved rule (this usually looks like `username:rule-name`).
```

## Privacy and Security

> ### Where do you store data?

Reviewer: ```suggestion
### Where does r2c store data?
```

Maybe we can have question here. But one which shows whom are we asking.
